### PR TITLE
ports/rp2/machine_uart.c: fix handling of serial break condition

### PR DESCRIPTION
**Pull request by mvds00**
The FIFO reports not only the bytes read, but also 4 error bits. These were not checked, leading to NUL value read in case of break and possible garbage bytes being written on parity/framing error.

This patch addresses the issue that NUL bytes are incorrectly read on break, and at least provides the boilerplate code and comments for error handling, that may be implemented in the future. This patch only addresses the rp2 port, but it is likely other ports are also affected by this bug. I am willing to look into that once this PR is accepted.

Note that there are interesting things that can be done with break conditions (e.g. automatically splitting read data on them and/or signalling break to data consuming code), but the first thing is to have at least these error flags checked.

### CODE REVIEW

File: ports/rp2/machine_uart.c [hunk 1]
```diff
@@ -140,8 +140,27 @@ static inline void read_mutex_unlock(machine_uart_obj_t *u) {
...
+                // read from dr.
+                continue;
+            }
+            if (c & UART_UARTDR_PE_BITS) {
+                // Parity Error: The byte we read is invalid.
+            }
+            if (c & UART_UARTDR_FE_BITS) {
+                // Framing Error: We did not receive a valid stop bit.
+            }
+
```
Comment (robert-hh) [inline ports/rp2/machine_uart.c:162]: I assume that the compiler will eliminate all error bit tests besides the one for break.

**Comment by thingsconnected** [comment]
@robert-hh are there issues that need to be addressed?

**Comment by robert-hh** [comment]
You have to wait, until the maintainer has time for it. That may take a while.

**Comment by dpgeorge** [comment]
This looks good to me.  A break condition on the UART line is definitely not the same as a NUL character, so the break should be handled differently.

**Pull request merged**